libr/fs/p/fs_r2.c

```diff
@@ -16,14 +16,14 @@ typedef struct {
 	WriteHandler write;
 } Routes;
 
-static RFSFile *__flags_cat(RFSRoot *root, RFSFile *file, const char *path);
-static RFSFile *__cfg_cat(RFSRoot *root, RFSFile *file, const char *path);
-static RFSFile *__seek_cat(RFSRoot *root, RFSFile *file, const char *path);
-static RFSFile *__bsize_cat(RFSRoot *root, RFSFile *file, const char *path);
-static bool __cfg_write(RFSFile *file, ut64 addr, const ut8 *data, int len);
-static bool __seek_write(RFSFile *file, ut64 addr, const ut8 *data, int len);
-static bool __bsize_write(RFSFile *file, ut64 addr, const ut8 *data, int len);
-static RFSFile *__version(RFSRoot *root, RFSFile *file, const char *path);
+static int __flags_cat(RFSRoot *root, RFSFile *file, const char *path);
```

Comment:
the flags_cat function accepts file argument to be NULL, in this case it reutrns that reference. this "feature" is lost in here so i assume some regressions may happen after this change. it will be better to add r_return_val_if_fail() precondition in those functions to avoid passing NULL

actually in all those cat functions if this argument is null is always calling 		file = r_fs_file_new (root, path);
so i think we can simplify that by justmaking this line in the callers. which should be less